### PR TITLE
Add retry limit to ScoreAsync

### PR DESCRIPTION
**Pull request by csteegz**
Add a retry limit to ScoreAsync.

### CODE REVIEW

File: sample-clients/csharp/client.tests/ScoringClientTests.cs [hunk 1]
```diff
@@ -90,6 +91,43 @@ public async Task Retries_transient_exceptions(StatusCode transientStatusCode)
...
+
+            outputTensorProto.TensorShape = new TensorShapeProto();
+            outputTensorProto.TensorShape.Dim.Add(new TensorShapeProto.Types.Dim());
+            outputTensorProto.TensorShape.Dim[0].Size = 3;
+
+            var predictRequest = new PredictRequest();
+            var predictResponse = new PredictResponse();
+            predictResponse.Outputs.Add("output_alias", outputTensorProto);
+
+            var predictionServiceClientMock = new Mock<IPredictionServiceClient>();
```
Comment (petrodeg) [inline sample-clients/csharp/client.tests/ScoringClientTests.cs]: Maybe better to move creation of mocks to ctor and only leave mock configuration in here

File: sample-clients/csharp/client/ScoringClient.cs [hunk 2]
```diff
@@ -41,19 +42,20 @@ public ScoringClient(string host, int port, bool useSsl = false, string authKey
             _client = new PredictionServiceClientWrapper(new PredictionService.PredictionServiceClient(channel));
         }
 
-        public async Task<float[]> ScoreAsync(IScoringRequest request)
+        public async Task<float[]> ScoreAsync(IScoringRequest request, int retryCount = 20)
```
Comment (petrodeg) [inline sample-clients/csharp/client/ScoringClient.cs]: Why is it so high?

File: sample-clients/csharp/client/ScoringClient.cs [hunk 3]
```diff
@@ -41,19 +42,20 @@ public ScoringClient(string host, int port, bool useSsl = false, string authKey
...
         {
-            return await ScoreAsync<float[]>(request);
+            return await ScoreAsync<float[]>(request, retryCount);
         }
 
-        public async Task<T> ScoreAsync<T>(IScoringRequest request) where T : class
+        public async Task<T> ScoreAsync<T>(IScoringRequest request, int retryCount = 20) where T : class
```
Comment (petrodeg) [inline sample-clients/csharp/client/ScoringClient.cs]: Maybe it should be a const

File: sample-clients/csharp/client.tests/ScoringClientTests.cs (hunk 1 above)
Comment (csteegz) [reply sample-clients/csharp/client.tests/ScoringClientTests.cs]: Refactoring client/Mocks into another method makes sense - imo outside the scope of this PR.

File: sample-clients/csharp/client/ScoringClient.cs (hunk 2 above)
Comment (csteegz) [reply sample-clients/csharp/client/ScoringClient.cs]: Set in #14 -  I have no idea why 20 was picked. It being so high is why this needs to exist - some of our tests take a long time to time out.

File: sample-clients/csharp/client/ScoringClient.cs (hunk 3 above)
Comment (csteegz) [reply sample-clients/csharp/client/ScoringClient.cs]: C# doesn't have const parameters? Are you suggesting that retry count shouldn't be an optional parameter?

Comment (petrodeg) [reply sample-clients/csharp/client/ScoringClient.cs]: Nope. Just make a value to be a const
```c#
private const DefaultRetryCount  = 20
…
int retryCount = DefaultRetryCount 
```

Comment (csteegz) [reply sample-clients/csharp/client/ScoringClient.cs]: Oh - yeah it probably makes more sense as a member.

**Comment by csteegz** [comment]
It turns out GRPC already has backoff built in - https://github.com/grpc/grpc/commit/1edfb952d0aa12bdc154e85eab24e87621292366#diff-43617ed09eee9fdea5a5d14dd0f80281

Which explains why the test with 20 retries was failing in ~20 minutes instead of in the ~2 minutes we'd expect from just the backoff we have. I've removed our backoff.

**Pull request merged**